designdocs/wallettool-subcommands.adoc

```diff
@@ -0,0 +1,575 @@
+
+==== **Usage**:
+
+[source,bash]
+----
+wallet-tool create [--net <NETWORK>] [--seed <SEED> | --watchkey <XPUB>] [--date <DATE> | --unixtime <UNIX_TIMESTAMP>] [--output-script-type <TYPE>] [--force] <WALLET_FILE>
+----
+
+==== **Options**:
+- `--seed <SEED>`: Specifies a mnemonic code or raw seed in hex/base58 raw seed bytes.
+- `--watchkey <XPUB>`: If present, it creates a watching wallet using the specified base58 xpub.
+- `--date <DATE>`: Wallet creation date (formatted as `YYYY-MM-DD`).
+- `--unixtime <UNIX_TIMESTAMP>`: Wallet creation time in Unix timestamp format.
+- `--output-script-type <TYPE>`: If specified, use that for deriving addresses. Provide an output script type. Valid values: P2PKH, P2WPKH. Default: P2WPKH.
+- `--force`: Overwrites any existing wallet file.
+- `--net <NETWORK>`: Specifies the network.
+
+==== **Examples**:
+
+[arabic]
+. Create a wallet with a mnemonic seed:
++
+[source,bash]
+----
+wallet-tool create --seed "12 mnemonic words" --output-script-type P2WPKH newwallet.wallet
+----
+. Create a watch-only wallet:
++
+[source,bash]
+----
+wallet-tool create --watchkey xpub6xxxxxxxxxx watcher.wallet
+----
+. Set a wallet creation date and derive P2PKH addresses:
++
+[source,bash]
+----
+wallet-tool create --date 2022-01-01 --output-script-type P2PKH datedwallet.wallet
+----
+
+=== *`dump`*
+
+*Description*: Loads and prints the given wallet in textual form to stdout. Allows printing private keys, seeds, and unused lookahead keys if specified.
+
+==== *Usage*:
+
+[source,bash]
+----
+wallet-tool dump [--dump-privkeys] [--dump-lookahead] [--password <PASSWORD>] <WALLET_FILE>
+----
+
+==== *Options*:
+* `--dump-privkeys`: Displays wallet seed and private keys (password required for an encrypted wallet).
+* `--dump-lookahead`: Includes lookahead keys (pregenerated but unused).
+* `--password <PASSWORD>`: Password to decrypt and access private keys. For an encrypted wallet, the password is provided here.
+
+==== *Examples*:
+
+[arabic]
+. Dump wallet data without private keys:
++
+[source,bash]
+----
+wallet-tool dump example.wallet
+----
+. Dump wallet data including lookahead keys and private keys:
++
+[source,bash]
+----
+wallet-tool dump --dump-privkeys --dump-lookahead --password myPassword  example.wallet
+----
+
+=== *`upgrade`*
+
+*Description*: Upgrade basic or deterministic wallets to deterministic wallets of the given script type. If specified, uses a target script type for deriving new addresses.
+
+==== *Usage*:
+
+[source,bash]
+----
+wallet-tool upgrade [--output-script-type <TYPE>] <WALLET_FILE>
+----
+
+==== *Options*:
+* `--output-script-type <TYPE>`: If specified, use that as the upgrade target. Script type for address derivation. Note that upgrade can be made from `Basic` --> `P2PKH` --> `P2WPKH`.
```

Comment:
I have to investigate, but I'm not sure if the Basic to Deterministic migration still exists. I believe not.

In this case I'll fix the *current* documentation.

Reply:
Maybe related: eb15ac07d356af74216c86f969eb7f48cc032045 and 4e8a19997d4c08d3e2633cf9368d0cef9b6f7ba7

Reply:
I've put up this PR: https://github.com/bitcoinj/bitcoinj/pull/3842

I lean towards merging it first (and backport to 0.17) and then adapting this PR (very simple changes).